packages/mongodb-chatbot-server/src/routes/responses/createResponse.test.ts

```diff
@@ -544,197 +621,328 @@ describe("POST /responses", () => {
-    const response = await makeCreateResponseRequest({
-      previous_response_id: "123456789012123456789012",
-      store: false,
+      expect(response.status).toBe(200);
+      testInvalidResponses({
+        responses: results,
+        message: ERR_MSG.STORE_NOT_SUPPORTED,
+      });
     });
 
-    expect(response.statusCode).toBe(400);
-    expect(response.body.error).toEqual(
-      badRequestError(ERR_MSG.STORE_NOT_SUPPORTED)
-    );
-  });
-
-  it("Should return 400 if `store: true` and `storeMessageContent: false`", async () => {
-    const conversation =
-      await appConfig.conversationsRouterConfig.conversations.create({
+    it("Should return 400 if `store: true` and `storeMessageContent: false`", async () => {
+      const initialMessages: Array<SomeMessage> = [
+        { role: "user", content: "Initial message!" },
+      ];
+      const { messages } = await conversations.create({
         storeMessageContent: false,
-        initialMessages: [{ role: "user", content: "" }],
+        initialMessages,
       });
 
-    const previousResponseId = conversation.messages[0].id.toString();
-    const response = await makeCreateResponseRequest({
-      previous_response_id: previousResponseId,
-      store: true,
-    });
+      const previous_response_id = messages[messages.length - 1].id.toString();
+      const { response } = await makeClientAndRequest({
+        previous_response_id,
+        store: true,
+      });
+      const results = await collectStreamingResponse(response);
 
-    expect(response.statusCode).toBe(400);
-    expect(response.body.error).toEqual(
-      badRequestError(ERR_MSG.CONVERSATION_STORE_MISMATCH)
-    );
+      expect(response.status).toBe(200);
+      testInvalidResponses({
+        responses: results,
+        message: ERR_MSG.CONVERSATION_STORE_MISMATCH,
+      });
+    });
   });
 });
 
 // --- HELPERS ---
 
-const badRequestError = (message: string) => ({
-  type: ERROR_TYPE,
-  code: ERROR_CODE.INVALID_REQUEST_ERROR,
+const getMessageIdFromResults = (results?: Array<any>) => {
+  if (!results?.length) throw new Error("No results found");
+
+  const messageId = results[results.length - 1]?.response?.id;
+
+  if (typeof messageId !== "string") throw new Error("Message ID not found");
+
+  return new ObjectId(messageId);
+};
+
+interface TestInvalidResponsesParams {
+  responses: Array<any>;
+  message: string;
+}
+
+const testInvalidResponses = ({
+  responses,
   message,
-});
+}: TestInvalidResponsesParams) => {
+  expect(Array.isArray(responses)).toBe(true);
+  expect(responses.length).toBe(1);
+  expect(responses[0]).toEqual({
+    ...formatOpenAIStreamError(400, ERROR_CODE.INVALID_REQUEST_ERROR, message),
+    sequence_number: 0,
+  });
+};
+
+interface TestResponsesParams {
+  requestBody: Partial<CreateResponseRequest["body"]>;
+  responses: Array<any>;
+}
+
+const testResponses = ({ requestBody, responses }: TestResponsesParams) => {
```

Comment:
nit cont.: also change this name to `expectValidResponses()` for consistency/clarity